checker/match.go

```diff
@@ -174,3 +177,47 @@ func matchHTTPMethods(methods []string, reqMethod string) bool {
 	log.Debug("HTTP Method not matched.")
 	return false
 }
+
+func matchL4Protocol(rule *proto.Rule, dest *authz.AttributeContext_Peer) bool {
+	// Extract L4 protocol type of socket address for destination peer context. Match against rules.
+	if dest == nil {
+		log.Warn("nil request destination peer.")
+		return false
+	}
+
+	// default protocol is TCP
+	reqProtocol := "TCP"
+	if a := dest.GetAddress(); a != nil {
+		if sa := a.GetSocketAddress(); sa != nil {
+			reqProtocol = sa.GetProtocol().String()
+		}
+	}
+
+	checkStringInRuleProtocol := func(p *proto.Protocol, s string) bool {
+		// Check if given protocol string matches what is specified in rule.
+		if name := p.GetName(); name != "" {
+			return strings.ToLower(name) == strings.ToLower(s)
+		}
+
+		// Envoy supports TCP only. Add a k:v into the map if more protocol is supported in the future.
+		protocolMap := map[int32]string{6: "TCP"}
+		if name, ok := protocolMap[p.GetNumber()]; ok {
+			return strings.ToLower(name) == strings.ToLower(s)
```

Comment:
Feels like a waste to define your map in uppercase, just to convert it to lowercase 2 lines later.

Reply:
Fixed